x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/search_and_filter_bar.tsx

```diff
@@ -328,63 +376,22 @@ export const SearchAndFilterBar: React.FunctionComponent<{
                 </EuiFilterButton>
               </EuiFilterGroup>
             </EuiFlexItem>
-            {selectedAgents.length === 0 && (
-              <EuiFlexItem>
-                <EuiToolTip
-                  content={
-                    <FormattedMessage
-                      id="xpack.fleet.agentList.addFleetServerButton.tooltip"
-                      defaultMessage="Fleet Server is a component of the Elastic Stack used to centrally manage Elastic Agents"
-                    />
-                  }
-                >
-                  <EuiButton onClick={onClickAddFleetServer} data-test-subj="addFleetServerButton">
-                    <FormattedMessage
-                      id="xpack.fleet.agentList.addFleetServerButton"
-                      defaultMessage="Add Fleet Server"
-                    />
-                  </EuiButton>
-                </EuiToolTip>
+            {(selectionMode === 'manual' && selectedAgents.length) ||
+            (selectionMode === 'query' && totalAgents > 0) ? (
```

Comment:
small refactor to move the visibility condition of bulk action button to a more logical place, this fixes a small issue of UI elements were not aligned (Add agent and Upgrade available box below)
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/90178898/192772133-febbfb5c-9427-4d03-888e-2f58264fb07f.png">